spec/controllers/credentials_controller_spec.rb

```diff
@@ -2,39 +2,59 @@
 
 require 'spec_helper'
 
-describe CredentialsController, :type => :controller do
+describe CredentialsController, :type => :request do
+  before(:all) { Slosilo["authn:rspec"] ||= Slosilo::Key.new }
+
   let(:login) { "u-#{random_hex}" }
   let(:account) { "rspec" }
   let(:authenticator) { "authn" }
+  let(:authenticate_url) do
+    "/#{authenticator}/#{account}/#{login}/authenticate"
+  end
+  let(:login_url) do
+    "/#{authenticator}/#{account}/login"
+  end
+  let(:update_password_url) do
+    "/authn/#{account}/password"
+  end
+  let(:update_api_key_url) do
+    "/#{authenticator}/#{account}/api_key"
+  end
   
   context "#rotate_api_key" do
+
     shared_examples_for "authentication denied" do
       it "is unauthorized" do
-        post :rotate_api_key, account: account, authenticator: authenticator
+        put(update_api_key_url, env: request_env)
         expect(response.code).to eq("401")
       end
     end
     
     context "without auth" do
+      let(:request_env) { {} } # empty hash
       it_should_behave_like "authentication denied"
     end
+
     context "with token auth" do
       include_context "create user"
       include_context "authenticate Token"
       it_should_behave_like "authentication denied"
     end
+
     context "with basic auth" do
       include_context "authenticate Basic"
+
       context "user not found" do
         let(:basic_password) { "the-password" }
         it_should_behave_like "authentication denied"
       end
+
       context "on valid user" do
         let(:basic_password) { api_key }
         include_context "create user"
+
         it "rotates the user's API key" do
-          api_key = the_user.api_key
-          post :rotate_api_key, account: account, authenticator: authenticator
+          put(update_api_key_url, env: request_env)
```

Comment:
Is this a change in the API functionality? (post -> put)

Reply:
I don't believe so.  I think it was a case of mis-naming before, or else just a change to the test itself.

Reply:
🤞